<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ScheduledInvestment</class>
 <widget class="QMainWindow" name="ScheduledInvestment">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>710</width>
    <height>395</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>定投组合表现</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="spacing">
       <number>5</number>
      </property>
      <item>
       <widget class="QLabel" name="lblStartDate">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>开始日期</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDateEdit" name="dteStartDate">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="displayFormat">
         <string>yyyy-MM-dd</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblInitialInvestment">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>初始总金额</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spbInitialInvestment">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="suffix">
         <string>万元</string>
        </property>
        <property name="prefix">
         <string/>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblEachInvestment">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>每期总金额</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spbEachInvestment">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="suffix">
         <string>万元</string>
        </property>
        <property name="prefix">
         <string/>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblInvestInterval">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>定投间隔</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spbInvestInterval">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="suffix">
         <string>月</string>
        </property>
        <property name="prefix">
         <string>每</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="value">
         <number>3</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <property name="spacing">
       <number>5</number>
      </property>
      <item>
       <widget class="QCheckBox" name="cbxSmartBuy">
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>以</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spbSmartBuyMaPeriod">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="suffix">
         <string>月</string>
        </property>
        <property name="minimum">
         <number>3</number>
        </property>
        <property name="maximum">
         <number>60</number>
        </property>
        <property name="value">
         <number>6</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblSmartBuyMa">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>均线之上</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spbSmartBuyMaBias">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="suffix">
         <string>%</string>
        </property>
        <property name="minimum">
         <number>-20</number>
        </property>
        <property name="maximum">
         <number>20</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblSmartBuyExplanation">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>的价格为基准，每上涨或下跌1%，相应减少或增加买入投资额 × </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spbSmartBuyFactor">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="suffix">
         <string>%</string>
        </property>
        <property name="prefix">
         <string/>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>20</number>
        </property>
        <property name="value">
         <number>5</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QCheckBox" name="cbxSmartSell">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>比上期上涨达到</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spbSmartSellThreshold">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="suffix">
         <string>%</string>
        </property>
        <property name="minimum">
         <number>10</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblSmartSellExplanation">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>则不买入并卖出一笔，每额外上涨1%多卖出投资额 ×</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spbSmartSellFactor">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="suffix">
         <string>%</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>20</number>
        </property>
        <property name="value">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbxBuyBackAfterSell">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>卖出后若下期下跌则买回</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <widget class="QCheckBox" name="cbxBuyByPercent">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>单月跌幅每达到</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spbBuyByPercent">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="suffix">
         <string>%</string>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblBuyByPercent">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>立刻买入一笔</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbxSellByPercent">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>单月涨幅每达到</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spbSellByPercent">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="suffix">
         <string>%</string>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="value">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblSellByPercent">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>立刻卖出一笔</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnImportPlan">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>导入策略</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnExportPlan">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>导出策略</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>5</number>
      </property>
      <item>
       <widget class="QLabel" name="lblNewStockCode">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>股票代码/名称</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="iptStockCode">
        <property name="maximumSize">
         <size>
          <width>70</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>贵州茅台</string>
        </property>
        <property name="maxLength">
         <number>6</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnAddStock">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>添加股票</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnRemoveStock">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>删除股票</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnClearStocks">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>清空列表</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnImportStockList">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>导入列表</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnExportStockList">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>导出列表</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnAutoSplitRatio">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>平分仓位</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTableWidget" name="tblStockList">
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="sortingEnabled">
       <bool>true</bool>
      </property>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>80</number>
      </attribute>
      <column>
       <property name="text">
        <string>股票代码</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>股票名称</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>持股仓位</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>上市日期</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>年化复利</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>累计投入</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>最终获利</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>最高亏损</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>盈亏比例</string>
       </property>
      </column>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <widget class="QPushButton" name="btnStartInvesting">
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>开始定投</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblTradeSummary">
        <property name="font">
         <font>
          <family>Helvetica</family>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>共计成本0元，至今获利0元，收益率0%</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>710</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>btnExportPlan</sender>
   <signal>clicked()</signal>
   <receiver>ScheduledInvestment</receiver>
   <slot>export_investment_plan()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>699</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>240</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnImportPlan</sender>
   <signal>clicked()</signal>
   <receiver>ScheduledInvestment</receiver>
   <slot>import_investment_plan()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>618</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>354</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnExportStockList</sender>
   <signal>clicked()</signal>
   <receiver>ScheduledInvestment</receiver>
   <slot>export_stock_list()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>602</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnImportStockList</sender>
   <signal>clicked()</signal>
   <receiver>ScheduledInvestment</receiver>
   <slot>import_stock_list()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>514</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>562</x>
     <y>82</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnAddStock</sender>
   <signal>clicked()</signal>
   <receiver>ScheduledInvestment</receiver>
   <slot>add_stock_code()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>128</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnRemoveStock</sender>
   <signal>clicked()</signal>
   <receiver>ScheduledInvestment</receiver>
   <slot>remove_stock_code()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>336</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>287</x>
     <y>128</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnClearStocks</sender>
   <signal>clicked()</signal>
   <receiver>ScheduledInvestment</receiver>
   <slot>clear_stock_list()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>425</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>410</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnStartInvesting</sender>
   <signal>clicked()</signal>
   <receiver>ScheduledInvestment</receiver>
   <slot>start_investing()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>84</x>
     <y>384</y>
    </hint>
    <hint type="destinationlabel">
     <x>562</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tblStockList</sender>
   <signal>cellDoubleClicked(int,int)</signal>
   <receiver>ScheduledInvestment</receiver>
   <slot>show_stock_graph()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>319</x>
     <y>275</y>
    </hint>
    <hint type="destinationlabel">
     <x>568</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbxSmartBuy</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbSmartBuyFactor</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>17</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>646</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnAutoSplitRatio</sender>
   <signal>clicked()</signal>
   <receiver>ScheduledInvestment</receiver>
   <slot>auto_split_ratio()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>690</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>28</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbxSmartBuy</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbSmartBuyMaPeriod</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>22</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>86</x>
     <y>69</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbxSmartBuy</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbSmartBuyMaBias</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>38</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>70</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbxSmartSell</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbSmartSellThreshold</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>114</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>249</x>
     <y>101</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbxSmartSell</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbSmartSellFactor</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>51</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>519</x>
     <y>102</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbxSmartSell</sender>
   <signal>toggled(bool)</signal>
   <receiver>cbxBuyBackAfterSell</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>604</x>
     <y>101</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbxBuyByPercent</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbBuyByPercent</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>124</y>
    </hint>
    <hint type="destinationlabel">
     <x>158</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbxSellByPercent</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbSellByPercent</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>365</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>120</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>export_investment_plan()</slot>
  <slot>import_investment_plan()</slot>
  <slot>export_stock_list()</slot>
  <slot>import_stock_list()</slot>
  <slot>add_stock_code()</slot>
  <slot>remove_stock_code()</slot>
  <slot>clear_stock_list()</slot>
  <slot>start_investing()</slot>
  <slot>show_stock_graph()</slot>
  <slot>auto_split_ratio()</slot>
 </slots>
</ui>
